rust/src/dcerpc/dcerpc.rs

```diff
@@ -621,7 +605,7 @@ impl DCERPCState {
                         header.rpc_vers,
                         header.rpc_vers_minor
                     );
-                    return -1;
+                    return -3;
```

Comment:
Please add `-3` to the return values listed in the function header comments.

Reply:
I would even rather get rid of `process_header` to match directly nom  ` Err(Err::Incomplete(_)) ` to our `AppLayerResult::incomplete`

Reply:
But, this is a case when the data is there but is not compliant w the dcerpc standards so likely not dcerpc. Do you think that the condition checks for values of the necessary struct items should also happen in the nom parsers?

Reply:
That is not what I mean.

I mean remove the definition of `process_header` and use directly its code :

ie instead of `parsed = self.process_header(cur_i);`a few lines below have

```
match parser::parse_dcerpc_header(cur_i) {
Ok((leftover_bytes, header)) => {
                if header.rpc_vers != 5
                    || (header.rpc_vers_minor != 0 && header.rpc_vers_minor != 1)
                {
    return AppLayerResult::err();
}
Err(Err::Incomplete(_)) => {
    return AppLayerResult::incomplete(0, DCERPC_HDR_LEN as u32);
}
```

Reply:
oh. I see! Thank you!